src/main/java/com/aws/greengrass/deployment/DefaultDeploymentTask.java

```diff
@@ -111,6 +114,20 @@ public DeploymentResult call() throws InterruptedException {
                     .kv("Deployment service config", deploymentServiceConfig.toPOJO().toString())
                     .log("Starting deployment task");
 
+            // Enforce deployment sequence validation using timestamps:
+            // If we have already processed a more recent deployment for a thing group, then we will drop
+            // this deployment and only process it if it is newer (or same) deployment for that thing group.
+            if (isStaleDeployment(deploymentDocument)) {
```

Comment:
can we do this in a different PR. Stale deployments and subgroups don't need to be tied.

Reply:
That would be tough now!

Reply:
I have split the commit and removed the stale deployment rejection. I will publish a separate CR for that.